addon-resizer/nanny/kubernetes_client.go

```diff
@@ -57,11 +60,35 @@ type kubernetesClient struct {
 // 2) using etcd_object_count metric exposed by kube-apiserver
 func (k *kubernetesClient) CountNodes() (uint64, error) {
 	if k.useMetrics {
-		return k.countNodesThroughMetrics()
+		return k.countResourcesThroughMetrics(nodeResourceName)
 	}
 	return k.countNodesThroughAPI()
 }
 
+// CountContainers returns the number of containers in the cluster:
+// 1) by listing Pods (excluding terminated pods) using API (default) and calculating the sum of initContainers, ephemeralContainers, and containers.
+// 2) using etcd_object_count metric exposed by kube-apiserver
+func (k *kubernetesClient) CountContainers() (uint64, error) {
+	if k.useMetrics {
+		return k.countResourcesThroughMetrics(podResourceName)
+	}
+	return k.countContainersThroughAPI()
+}
+
+func (k *kubernetesClient) countContainersThroughAPI() (uint64, error) {
+	// Set ResourceVersion = 0 to use cached versions.
+	options := metav1.ListOptions{
+		ResourceVersion: "0",
+		FieldSelector:   getPodSelectorExcludingDonePodsOrDie(),
+	}
+	pods, err := k.clientset.CoreV1().Pods("").List(context.TODO(), options)
```

Comment:
Please use [NamespaceAll](https://github.com/kubernetes/autoscaler/blob/4cb926b2f0bb4e4b5fc206bf5680821311b09487/addon-resizer/vendor/k8s.io/api/core/v1/types.go#L30C2-L30C14) const to make this easier to understand

```suggestion
	pods, err := k.clientset.CoreV1().Pods(corev1.NamespaceAll).List(context.TODO(), options)
```

Reply:
Addressed.